File: py/33.py
# 100
import math
import euler
D = 2
N = 10 ** D
np = 1
dp = 1
for n in xrange(N/10, N):
    for d in xrange(n+1, N):
        nn = euler.digit_usage(n)
        dd = euler.digit_usage(d)
        c = nn & dd
        if c == 0 or c == 1 or math.log(c, 2) % 1.0 != 0 or \
                nn ^ c == 0 or dd ^ c == 0:
            continue
        n2 = math.log(nn ^ c, 2)
        d2 = math.log(dd ^ c, 2)
        if d2 != 0 and float(n) / d == n2 / d2:
            np *= n
            dp *= d
np, dp = euler.reduce_fraction(np, dp)
print dp

File: c/33.c
/* 100 */
#include <math.h>
#include <stdio.h>
#include "euler.h"

#define N 100

int main(int argc, char **argv)
{
    double cc, n2, d2;
    int n, nn, d, dd, c;
    int np = 1, dp = 1;
    for (n = N/10; n < N; n++)
    {
        for (d = n + 1; d < N; d++)
        {
            nn = digit_usage(n, 0);
            dd = digit_usage(d, 0);
            c = nn & dd;
            if (c == 0 || c == 1 || (nn ^ c) == 0 || (dd ^ c) == 0)
                continue;
            cc = log2(c);
            if (cc != floor(cc))
                continue;
            n2 = log2(nn ^ c);
            d2 = log2(dd ^ c);
            if ((double)n / d == n2 / d2)
            {
                np *= n;
                dp *= d;
            }
        }
    }

    n = 2;
    while (n < sqrt(dp))
    {
        while (np % n == 0 && dp % n == 0)
        {
            np /= n;
            dp /= n;
        }
        n++;
    }
    printf("%d\n", dp);
    return 0;
}
